### PR TITLE
4792 other occurrences4th column

### PR DESCRIPTION
**Pull request by wschaeferB**

### CODE REVIEW

File: Core/src/org/sleuthkit/autopsy/centralrepository/contentviewer/DataContentViewerOtherCases.java [hunk 1]
```diff
@@ -218,54 +231,43 @@ private void showCommonalityDetails() {
...
-                // query case details
-                CorrelationCase eamCase = dbManager.getCaseByUUID(eamCasePartial.getCaseUUID());
-                if (eamCase == null) {
-                    JOptionPane.showConfirmDialog(showCaseDetailsMenuItem,
-                            Bundle.DataContentViewerOtherCases_caseDetailsDialog_noDetails(),
-                            caseDisplayName,
-                            DEFAULT_OPTION, PLAIN_MESSAGE);
-                    return;
+                List<OtherOccurrenceNodeData> rowList = filesTableModel.getListOfNodesForFile(selectedRowModelIdx);
+                if (rowList.get(0) instanceof OtherOccurrenceNodeInstanceData) {
```
Comment (eugene7646) [inline Core/src/org/sleuthkit/autopsy/centralrepository/contentviewer/DataContentViewerOtherCases.java]: We should probably check that the list is not empty, before accessing the data?

File: Core/src/org/sleuthkit/autopsy/centralrepository/contentviewer/DataContentViewerOtherCases.java [hunk 2]
```diff
@@ -279,45 +281,45 @@ private void saveToCSV() throws NoCurrentCaseException {
...
-
-            // write column names
-            content = new StringBuilder("");
-            for (int colIdx = 0; colIdx < colCount; colIdx++) {
-                content.append('"').append(tableModel.getColumnName(colIdx)).append('"');
-                if (colIdx < (colCount - 1)) {
-                    content.append(",");
+            //write headers 
+            StringBuilder headers = new StringBuilder("");
+            headers.append('"').append(Bundle.DataContentViewerOtherCasesModel_csvHeader_case()).append('"').append(',');
```
Comment (eugene7646) [inline Core/src/org/sleuthkit/autopsy/centralrepository/contentviewer/DataContentViewerOtherCases.java]: Codacy has a lot of issues with how StringBuilder is used here.

File: Core/src/org/sleuthkit/autopsy/centralrepository/contentviewer/OtherOccurrenceNodeInstanceData.java [hunk 3]
```diff
@@ -199,17 +216,37 @@ AbstractFile getAbstractFile() throws EamDbException {
...
     }
+
+    /**
+     * Create a string representation of the node's data comma separated with a
+     * line separator ending
+     *
+     * @return a comma separated string representation of the node's data
+     */
+    String toCsvString() {
+        StringBuilder line = new StringBuilder("");
```
Comment (eugene7646) [inline Core/src/org/sleuthkit/autopsy/centralrepository/contentviewer/OtherOccurrenceNodeInstanceData.java]: Codacy has a lot of flags regarding use of StringBuilder here.

File: Core/src/org/sleuthkit/autopsy/centralrepository/contentviewer/OtherOccurrencesFilesTableModel.java [hunk 4]
```diff
@@ -19,150 +19,65 @@
...
     @Override
     public String getColumnName(int colIdx) {
-        return TableColumns.values()[colIdx].columnName();
+        return Bundle.OtherOccurrencesFilesTableModel_fileName();
     }
 
     @Override
     public Object getValueAt(int rowIdx, int colIdx) {
-        if (0 == nodeDataList.size()) {
+        if (0 == nodeKeys.size()) {
```
Comment (eugene7646) [inline Core/src/org/sleuthkit/autopsy/centralrepository/contentviewer/OtherOccurrencesFilesTableModel.java]: Codacy says to use isEmpty() here instead

**Comment by eugene7646** [comment]
Most of Codacy flags appear to make sense so please take a look at them.

**Comment by wschaeferB** [comment]
Thank you, I am going to have possibly a couple commits here, I will comment again when they are all in.

**Comment by wschaeferB** [comment]
Codacy has been satisfied for now, and the checking of indexes and collection sizes should be adequate now.

**Pull request merged**